### ISSUE TITLE
Color ramp for LBDA dataset makes most valid pixels transparent, the same as NODATA.

### ISSUE THREAD

**Issue by tmcphillips** [openskope/Datasets#18]

**Comment by tmcphillips** [comment]
The values of the PMDI variable for the LBDA dataset is distributed around the value zero with a large fraction of values between -1 and 1 (see [Jupyter notebook)](https://github.com/openskope/skope-notebooks/blob/master/download_lbda_v2_dataset_from_noaa.ipynb):
![image](https://user-images.githubusercontent.com/3218259/41486441-65625710-7099-11e8-9f05-e3beb6e49c1b.png)

Meanwhile, the color ramp for this variable colors pixels between -1 as 1 as transparent in the web app:
![image](https://user-images.githubusercontent.com/3218259/41486365-0c58f94e-7099-11e8-9fc4-1c0f02c19688.png)

Regions with NODATA also are transparent, so there is no way to distinguish many of the valid pixels from regions where there is no data.  A color ramp that avoids this problem (same [Jupyter notebook](https://github.com/openskope/skope-notebooks/blob/master/download_lbda_v2_dataset_from_noaa.ipynb) as above) yields the view below of the same year (2017) as rendered above:

![image](https://user-images.githubusercontent.com/3218259/41486564-df338136-7099-11e8-849d-50ca8c317214.png)

This problem needs to be addressed before we release the LBDA data set to production.

**Comment by tmcphillips** [comment]
The GeoServer SLD REST service provides the Jet color ramp (the one used in the figure immediately above) as one of the predefined color ranges for raster data (the predefined ranges are RED, BLUE, GRAY, JET, RANDOM, and CUSTOM): 
http://docs.geoserver.org/stable/en/user/community/sldservice/index.html#classify-raster-data

The general definition of the Jet color ramp (as defined by MATLAB) can be found here...
https://bugs.launchpad.net/inkscape/+bug/236508
...according to which:
> It is defined as a linear ramp between the following colours: "#00007F", "blue", "#007FFF", "cyan", "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000"

**Comment by jterstriep** [comment]
The Jet colormap is not considered particularly good for visualization purposes.

https://jakevdp.github.io/blog/2014/10/16/how-bad-is-your-colormap/

Additional color maps from matplotlib.

https://bids.github.io/colormap/

**Comment by tmcphillips** [comment]
Yeah, I found that blog post, too. I am only arguing that it is visible, which beats invisible, and intensely colored throughout which contrasts well with the less intense colors that base maps often use.

**Comment by jterstriep** [comment]
Mostly, I was just documenting it because that particular link has some other color ramp options.

**Comment by tmcphillips** [comment]
Is everyone happy with the current default color ramp for the PMDI variable?  The colors corresponding to values between -3 and +3 are not very intense.  Accepting the default 50% opacity it's still pretty hard to see where there is and isn't data:

![image](https://user-images.githubusercontent.com/3218259/41945647-01e81008-7963-11e8-9d43-f274a1c8404f.png)

Manually adjusting to 100% opacity it's a bit better:

![image](https://user-images.githubusercontent.com/3218259/41945684-39bc8892-7963-11e8-9599-8dd68107c825.png)

**Comment by bocinsky** [comment]
Part of the issue might be the baseman we are using. How would people feel about switching to a grayscale basemap, like OpenStreetMap.BlackAndWhite? Some other options are at https://leaflet-extras.github.io/leaflet-providers/preview/ <https://leaflet-extras.github.io/leaflet-providers/preview/>.

http://{s}.tiles.wmflabs.org/bw-mapnik/{z}/{x}/{y}.png

**Comment by bocinsky** [comment]
Stamen.Toner is an even more neutral canvas.

**Comment by kintigh** [comment]
I think the base map we have now is a pretty good compromise between topography and landscape features and roads, land status,  political entities etc.  I think for trying to find your research area the B/W and stamen Toner ones would be impossible.  We have as an issue about offering options and letting people choose (and I am happy for these to be options) , but by default I think we need the topography and land status,  even if it makes the colors more difficult.

Keith
__________________________________________________________________
Keith W. Kintigh, Professor   kintigh@asu.edu<mailto:kintigh@asu.edu>   ASU Directory Page<https://webapp4.asu.edu/directory/person/56433>
ASU School of Human Evolution & Social Change<http://shesc.asu.edu/>
ASU Center for Archaeology & Society<https://shesc.asu.edu/centers/archaeology-and-society>, Co-director
Arizona State University<http://asu.edu/>, Box 872402, Tempe, AZ 85287-2402

Coalition for Archaeological Synthesis<http;/archsynth.org>, Interim Co-Chair
Journal of Computer Applications in Archaeology<http://journal.caa-international.org/>, Editorial Board

**Comment by bocinsky** [comment]
That’s a good point, Keith.
